main/src/ca/uwaterloo/flix/language/ast/ParsedAst.scala

```diff
@@ -1480,6 +1480,119 @@ object ParsedAst {
 
   }
 
+  /**
+    * Effect Set
+    */
+  sealed trait EffectSet
+
+  object EffectSet {
+    /**
+      * Singleton effect set.
+      *
+      * @param sp1 the position of the first character in the set.
+      * @param eff the effect.
+      * @param sp2 the position of the last character in the set.
+      */
+    case class Singleton(sp1: SourcePosition, eff: Effect, sp2: SourcePosition) extends EffectSet
+
+    /**
+      * Pure effect set.
```

Comment:
Represents the empty effect set. Written as `{ }` or `{ Pure }` depending on the syntactic context.